src/include/dawg/error.h

```diff
@@ -0,0 +1,139 @@
+    /*  Dawg - DNA Assembly with Gaps - Simulating Sequence Evolution
+    Copyright (c) 2018 Juan J. Garcia Mesa
+    This program is free software; you can redistribute it and/or modify
+    it under the terms of the GNU General Public License as published by
+    the Free Software Foundation; either version 2 of the License, or
+    (at your option) any later version.
+    This program is distributed in the hope that it will be useful,
+    but WITHOUT ANY WARRANTY; without even the implied warranty of
+    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+    GNU General Public License for more details.
+    You should have received a copy of the GNU General Public License
+    along with this program; if not, write to the Free Software
+    Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA  02111-1307  USA
+*/
+
+#pragma once
+#ifndef DAWG_ERROR_H
+#define DAWG_ERROR_H
+
+#include <string>
+#include <iostream>
+#include <system_error>
+
+// custom error code enum
+enum class dawg_error {
+    // i/o					//object needed to display error message
+    open_input_file_fail = 10,			//const char*
+    open_output_file_fail = 11,
+    unknown_output_format = 12,			//std::string
+    parse_input_file_fail = 13,			//const char*
+    parsing_failed = 14,
+
+    // indel model
+    indel_model_no_type = 20,
+
+    // configuration
+    bad_configuration = 30,
+    configuration_section_fail = 31,		//first->name (?)
+
+    // section
+    section_not_found = 40,			//std::string, std::string
+    section_already_specified = 41,		//std::string
+    section_sequence_type_invalid = 42,
+
+    // miscellaneous
+    invalid_genetic_code = 51,
+    invalid_sequence_type = 52,
+    creation_fail = 53,
+    invalid_value = 54,
+    model_no_name = 55,
+    segment_extra_root = 56,
+    no_sequence = 57,
+    invalid_tree = 58,
+    unexpected = 59,
+    param_missing = 60,
+    invalid_user_sequence = 61,
+};
+
+namespace std {
+    template <>
+	struct is_error_code_enum<dawg_error> : public true_type {};
+}// end namespace std
+
+namespace dawg {
+
+    extern std::string DAWG_ERROR_INFO_;
+
+    class dawg_error_category : public std::error_category {
+    public:
+
+	// return short descriptive name for the category
+	virtual const char *name() const noexcept override final {return "Dawg error category"; }
+	// return enum description
+	// TODO: add parameter(s) when needed for error message
+	virtual std::string message(int ev) const override final {
```

Comment:
DAWG_ERROR_INFO shouldn't be include here. Instead it should be accessed via a function. If the caller wants to output a more descriptive message.